packages/@yoda/bluetooth/hfp.js

```diff
@@ -0,0 +1,295 @@
+ * @property {string} HFP - Bluetooth hands-free profile.
+ */
+
+/**
+ * @typedef {object} RADIO_STATE
+ * @property {string} ON - Bluetooth radio state is ON.
+ * @property {string} OFF - Bluetooth radio state is OFF.
+ * @property {string} ON_FAILED - Turn on bluetooth radio failed.
+ */
+
+/**
+ * @typedef {object} CONNECTION_STATE
+ * @property {string} CONNECTED - Successfully connected to remote device.
+ * @property {string} DISCONNECTED - Disconnected from remote deivce.
+ * @property {string} CONNECT_FAILED - Failed to connect to remote device.
+ * @property {string} AUTOCONNECT_FAILED - Auto connection to history paired device failed after turn on bluetooth.
+ */
+
+/**
+ * When bluetooth's radio state is changed, such as on/off.
+ * @event module:@yoda/bluetooth.BluetoothHfp#radio_state_changed
+ */
+
+/**
+ * When bluetooth's connection state is changed, such as connected/disconnected.
+ * @event module:@yoda/bluetooth.BluetoothHfp#connection_state_changed
+ */
+
+/**
+ * When bluetooth's call state is changed, such as idle/incoming.
+ * @event module:@yoda/bluetooth.BluetoothHfp#call_state_changed
+ */
+
+/**
+ * Use `bluetooth.getAdapter(PROFILE.HFP)` instead of this constructor.
+ * @memberof module:@yoda/bluetooth
+ * @constructor
+ * @extends EventEmitter
+ * @param {string} deviceName - The device name.
+ */
+class BluetoothHfp extends EventEmitter {
+  constructor (deviceName) {
+    super()
+    this.lastMsg = {
+      'hfpstate': 'invalid', // opened, open failed, closing, closed, invalid.
+      'connect_state': 'invalid', // connecting, connected, connect failed, disconnecting, disconnected, invalid.
+      'connect_address': null,
+      'connect_name': null,
+      'service': 'active', // inactive, active.
+      'call': 'inactive', // inactive, active.
+      'setup': 'none', // none, incoming, outgoing, alerting.
+      'held': 'none', // none, hold_active, hold.
+      'audio': 'off' // on, off.
+    }
+    this._end = false
+    this.localName = deviceName // local bluetooth device name
+
+    this._flora = new FloraComp(logger)
+    this._flora.handlers = {
+      'bluetooth.hfp.event': this._onEvent.bind(this)
+    }
+    this._flora.init(null, {
+      'uri': 'unix:/var/run/flora.sock',
+      'bufsize': 40960,
+      'reconnInterval': 10000
+    })
+  }
+}
+
+/**
+ * @private
+ */
+BluetoothHfp.prototype.matchState = function (msg, filter) {
+  return (filter.hfpstate === msg.hfpstate || filter.hfpstate === undefined) &&
+    (filter.connect_state === msg.connect_state || filter.connect_state === undefined) &&
+    (filter.service === msg.service || filter.service === undefined) &&
+    (filter.call === msg.call || filter.call === undefined) &&
+    (filter.setup === msg.setup || filter.setup === undefined) &&
+    (filter.held === msg.held || filter.held === undefined) &&
+    (filter.audio === msg.audio || filter.audio === undefined)
+}
```

Comment:
If you are using `class` and `extends`, the following methods could be writtern in:

```js
class BluetoothHfp extends EventEmitter {
  constructor () {
  }
  matchState (msg, filter) {
  }
}
```

Reply:
Already moved private function into class body.